apps/dashboard/src/app/(dashboard)/(chain)/[chain_id]/[contractAddress]/modules/components/CurrencySelector.tsx

```diff
@@ -0,0 +1,190 @@
+import { Button } from "@/components/ui/button";
+import { FormControl } from "@/components/ui/form";
+import { Input } from "@/components/ui/input";
+import {
+  Select,
+  SelectContent,
+  SelectItem,
+  SelectTrigger,
+  SelectValue,
+} from "@/components/ui/select";
+import { CURRENCIES, type CurrencyMetadata } from "constants/currencies";
+import { useMemo, useState } from "react";
+import type { ControllerRenderProps, FieldValues } from "react-hook-form";
+import { NATIVE_TOKEN_ADDRESS, ZERO_ADDRESS, isAddress } from "thirdweb";
+import { useAllChainsData } from "../../../../../../../hooks/chains/allChains";
+
+interface CurrencySelectorProps<
+  TFieldValues extends FieldValues = FieldValues,
+> {
+  small?: boolean;
+  hideDefaultCurrencies?: boolean;
+  showCustomCurrency?: boolean;
+  isPaymentsSelector?: boolean;
+  defaultCurrencies?: CurrencyMetadata[];
+  contractChainId: number;
+  field: ControllerRenderProps<TFieldValues>;
+}
+
+export function CurrencySelector<
+  TFieldValues extends FieldValues = FieldValues,
+>({
+  small,
+  hideDefaultCurrencies,
+  showCustomCurrency = true,
+  isPaymentsSelector = false,
+  defaultCurrencies = [],
+  field,
+  contractChainId: chainId,
+}: CurrencySelectorProps<TFieldValues>) {
+  const { idToChain } = useAllChainsData();
+  const chain = chainId ? idToChain.get(chainId) : undefined;
```

Comment:
Don't use the `useAllChainsData` here and instead get the `chain` object from props so this component is more generic and can be used in other places 

also - move this inside the src/@/components/blocks folder